spec/models/accounting/qb/transaction_builder_spec.rb

```diff
@@ -1,111 +1,167 @@
+        private_note: memo,
+        description: description,
+        account: bank_account,
+        loan_transaction_type_value: :disbursement,
+        txn_date: date,
+        disbursement_type: disbursement_type,
+        check_number: check_number
+      )
+    end
+
+    let(:prin_line_item) {
+      build(:line_item,
+            account: principal_account,
+            posting_type: "Debit",
+            amount: 100)
+    }
+
+    let(:line_item_2) {
+      build(:line_item,
+            account: bank_account,
+            posting_type: "Credit",
+            amount: 25)
+    }
+
+    context "non-check disb" do
+      let(:disbursement_type) { :other }
+      it "sets payment_type to cash" do
+        p = subject.build_for_qb transaction
+        expect(p.payment_type).to eq "Cash"
+      end
+    end
+
+    it "calls create with correct data" do
+      expect(transaction.qb_object_type).to eq "Purchase"
+      p = subject.build_for_qb transaction
+
+      # disbursements exist in qb as purchases that have 1 li
+      # disbursements have 2 lis in madeline
+      # see purchase_extractor that adds the 2nd line item when a txn
+      # is imported to madeline from qb
+      expect(p.line_items.count).to eq 1
+      expect(p.private_note).to eq memo
+      expect(p.doc_number).to include "MS-Managed"
+      expect(p.doc_number).to include check_number
+      expect(p.txn_date).to eq date
+      li = p.line_items.first
+      expect(li.detail_type).to eq "AccountBasedExpenseLineDetail"
+      expect(li.id).to eq prin_line_item.qb_line_id
+      expect(li.amount).to eq transaction.amount
+      expect(li.description).to eq description
+      detail = li.account_based_expense_line_detail
+      expect(detail.account_ref.name).to eq principal_account.name
+      expect(detail.customer_ref.name).to eq customer.name
+      expect(detail.billable_status).to eq "NotBillable"
+    end
   end
 
-  context 'and date is supplied' do
-    let(:date) { 3.days.ago.to_date }
+  describe "a journal entry" do
+    before do
+      transaction.line_items << [line_item_1, line_item_2, line_item_3]
+    end
+    let(:transaction) do
+      Accounting::Transaction.create(
+        amount: amount,
+        customer: customer,
+        project: loan,
+        private_note: memo,
+        description: description,
+        account: bank_account,
+        loan_transaction_type_value: :repayment,
+        txn_date: date
+      )
+    end
+
+    let(:line_item_1) {
+      build(:line_item,
+            account: office_account,
+            posting_type: "Debit",
+            amount: 100)
+    }
+
+    let(:line_item_2) {
+      build(:line_item,
+            account: bank_account,
+            posting_type: "Credit",
+            amount: 25)
+    }
 
-    it 'creates JournalEntry with date' do
+    let(:line_item_3) {
+      build(:line_item,
+            account: principal_account,
+            posting_type: "Credit",
+            amount: 75)
+    }
```

Comment:
Can we fit these line items on one line each? If so, it might be a good idea since in my experience it makes it much easier to read the spec if it's not spread over many screens. I will sometimes even shorten var names or use a couple additional `let`s to make things fit on one line.